read-fonts/src/tables/truetype.rs

```diff
@@ -0,0 +1,68 @@
+//! Support for processing TrueType outlines.
+
+use std::fmt;
+
+use crate::{types::GlyphId, ReadError};
+
+mod deltas;
+mod mem;
+mod outline;
+mod scale;
+
+pub use mem::OutlineMemory;
+pub use outline::{HintOutline, Outline, OutlineInfo};
+pub use scale::Scaler;
+
+/// Recursion limit for processing composite outlines.
+///
+/// In reality, most fonts contain shallow composite graphs with a nesting
+/// depth of 1 or 2. This is set as a hard limit to avoid stack overflow
+/// and infinite recursion.
+pub const COMPOSITE_RECURSION_LIMIT: usize = 32;
+
+/// Number of phantom points generated at the end of an outline.
+pub const PHANTOM_POINT_COUNT: usize = 4;
+
+/// Errors that may occur when scaling glyphs.
+#[derive(Clone, Debug)]
+pub enum Error {
+    /// The requested glyph was not present in the font.
+    GlyphNotFound(GlyphId),
+    /// Exceeded a recursion limit when loading a glyph.
+    RecursionLimitExceeded(GlyphId),
+    /// Exceeded memory limits when loading a glyph.
+    InsufficientMemory,
```

Comment:
Shouldn't we know what glyph, what limit, and perhaps how much we needed?

Reply:
Unfortunately, the full context isn’t known at all places where this error is generated so we can’t provide useful information in the general case. Presumably, the user has computed the required size up front and provided a buffer >= that size (or returned their own error if unable to do so). If the buffer size does meet our requirement and we’re still failing then it’s a logic bug in our code and the only reasonable thing to do is file an issue.